packages/core/src/model-definition.ts

```diff
@@ -213,7 +215,7 @@ export class ModelDefinition {
         },
         globalOptions.define as ModelOptions,
       ),
-      modelOptions,
+      definedModelOptions,
```

Comment:
And with the previous change, you could eliminate the extra variable by removing undefined here